### PR TITLE
feat: editable columns in localStorage

### LINKED ISSUE #659: Make alerts table columns configureable
As a user, I would like to be able to chose what columns to be displayed in the view (per preset?).

### PR DESCRIPTION
**Pull request by talboren**
Closes #659 

## 📑 Description

- add props to `AlertTable` for hiding columns and ordering columns
- create AlertColumnsSelect component to select and order columns

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep-ui/app/alerts/alert-table.tsx [hunk 1]
```diff
@@ -217,14 +227,74 @@ export function AlertTable({
...
+  );
+
+  const [columns] = useState<typeof defaultColumns>(() => [
+    ...defaultColumns,
+    ...extraPayloadColumns,
+  ]);
+  const [columnOrder, setColumnOrder] = useState<ColumnOrderState>(
+    columnOrderLocalStorage ? JSON.parse(columnOrderLocalStorage) : []
+  );
+  const [columnVisibility, setColumnVisibility] = useState<{}>(
```
Comment (asharonbaltazar) [inline keep-ui/app/alerts/alert-table.tsx:268]: I think `react-table` exports a `Visibility` type

File: keep-ui/app/alerts/alert-table.tsx [hunk 2]
```diff
@@ -217,14 +227,74 @@ export function AlertTable({
     }),
   ];
 
+  const extraPayloadKeys = Array.from(
+    new Set(
+      alerts
+        .map((alert) => {
+          const extraPayload = getExtraPayloadNoKnownKeys(alert);
+          return Object.keys(extraPayload.extraPayload);
```
Comment (asharonbaltazar) [inline keep-ui/app/alerts/alert-table.tsx:235]: If you're having trouble with naming:

```suggestion
          const { extraPayload } = getExtraPayloadNoKnownKeys(alert);
          return Object.keys(extraPayload);
```

File: keep-ui/app/alerts/alert-table.tsx [hunk 3]
```diff
@@ -217,14 +227,74 @@ export function AlertTable({
...
+    getHiddenColumnsLocalStorageKey(presetName)
+  );
+  const columnOrderLocalStorage = localStorage.getItem(
+    getColumnsOrderLocalStorageKey(presetName)
+  );
+
+  const [columns] = useState<typeof defaultColumns>(() => [
+    ...defaultColumns,
+    ...extraPayloadColumns,
+  ]);
```
Comment (asharonbaltazar) [inline keep-ui/app/alerts/alert-table.tsx:264]: Is there a reason why `columns` has to be a state variable? I don't see it being manipulated anywhere

**Review by asharonbaltazar — APPROVED** [review]

File: keep-ui/app/alerts/alert-table.tsx (hunk 3 above)
Comment (talboren) [reply keep-ui/app/alerts/alert-table.tsx:264]: Not really, I kinda followed an example from react-table.

File: keep-ui/app/alerts/alert-table.tsx (hunk 1 above)
Comment (talboren) [reply keep-ui/app/alerts/alert-table.tsx:268]: True, VisibilityState!

**Pull request merged**